packages/shared/src/components/RichTextEditor/plugins/shared/LinkSelector/LinkSelector.tsx

```diff
@@ -0,0 +1,72 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { AppBridgeBlock } from '@frontify/app-bridge';
+import { Button, ButtonEmphasis, ButtonSize, ButtonStyle, ButtonType, IconLink, Modal } from '@frontify/fondue';
+import { useOverlayTriggerState } from '@react-stately/overlays';
+import React, { ReactElement, useEffect, useState } from 'react';
+import { DocumentLinks } from './DocumentLinks';
+
+type LinkSelectorProps = {
+    appBridge: AppBridgeBlock;
+    url: string;
+    onUrlChange: (value: string) => void;
+};
+
+export const LinkSelector = ({ appBridge, url, onUrlChange }: LinkSelectorProps): ReactElement => {
+    const { open: openLinkTree, isOpen: isLinkTreeOpen, close: closeLinkTree } = useOverlayTriggerState({});
+    const [selectedUrl, setSelectedUrl] = useState<string>(url);
+
+    const onSelectUrl = (url: string) => {
+        setSelectedUrl(url);
+    };
+
+    useEffect(() => {
+        if (url && !selectedUrl) {
+            setSelectedUrl(url);
+        }
+    }, [url, selectedUrl]);
+
+    return (
+        <div data-test-id="internal-link-selector">
+            <Button
+                icon={<IconLink />}
+                size={ButtonSize.Medium}
+                type={ButtonType.Button}
+                style={ButtonStyle.Default}
+                emphasis={ButtonEmphasis.Default}
+                onClick={() => openLinkTree()}
+            >
+                Internal link
+            </Button>
+            <Modal zIndex={200} onClose={() => closeLinkTree()} isOpen={isLinkTreeOpen} isDismissable>
```

Comment:
Modal Closes by click on it (not a link from the link tree), I think it should just close by click on "cancel" or by click on the "background"

Reply:
Fixed 👍🏻